### ISSUE TITLE
Random return 502

### ISSUE THREAD

**Issue by ksss** [matsumotory/ngx_mruby#268]
I'm encounering strange behavior.

```
upstream unicorn_server {
  server unix:/app/tmp/sockets/unicorn.sock fail_timeout=0;
}
server {
  location / {
    mruby_access_handler_code '
      req = Nginx::Request.new
      body = case req.method
      when "POST", "PUT"
        rc = req.read_body
        Nginx.log Nginx::LOG_ERR, "===== rc:#{rc} ====="
        req.get_body
      else
        ""
      end
      Nginx.log Nginx::LOG_ERR, "===== body:#{body} ====="
    ';
    proxy_pass http://unicorn_server;
  }
}
```

In almost every case, request success with this logs

```
===== rc:0 =====
===== body:{"hello": "ngx_mruby"} =====
```

But in very rare case, request failed with 502 and this logs

```
===== rc:-2 =====
===== body: =====
```

How should I handle in this case?
I want to send to the proxy app even failed case.

**Comment by matsumotory** [comment]
Thank you for your report. Could you give me the code that can be reproduced with only ngx_mruby instead of unicorn?

**Comment by ksss** [comment]
I tried to make a Dockerfile that can reproduce the problem.
But sorry, I couldn't reproduce on my local environment.

Strangely, This issue could only reproduce on my aws-ecs.

```
$ curl -X POST -H 'Content-type: application/json' -d '{"hello": "ngx_mruby"}' https://exampl.com/entry-point
```
=> CDN(aws-CloudFront) =>  ELB => docker contener on EC2 => nginx process => unicorn process

OS: alpine
nginx: v1.11.10
ngx_mruby: v1.18.10

**Comment by ksss** [comment]
@matsumotory I could represent return the `NGX_AGAIN`

https://gist.github.com/ksss/c743fa8d94d8702765c50064a914ed57

1. Applay tha patch `nginx.conf.patch`
1. Build docker
1. Run the container
1. Send slow request(see also `representer`)

But it couldn't represent segv.

**Comment by matsumotory** [comment]
fixed via #269

**Comment by matsumotory** [comment]
Peleased https://github.com/matsumotory/ngx_mruby/releases/tag/v1.19.0
